dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -3010,3 +3032,158 @@ IPropertyTree * RunProcess(XRefCmd cmd, unsigned nclusters,const char **clusters
     }
     return nullptr;
 }
+
+#ifdef _USE_CPPUNIT
+
+#include "unittests.hpp"
+
+#define CPPUNIT_ASSERT_EQUAL_STR(x, y) CPPUNIT_ASSERT_EQUAL(std::string(x ? x : ""),std::string(y ? y : ""))
+class DFUXrefLibTests : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE(DFUXrefLibTests);
+    CPPUNIT_TEST(TestParseFileName);
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+    void TestParseFileName()
+    {
+        StringBuffer mname;
+        unsigned num;
+        unsigned max;
+        unsigned stripeNum;
+        unsigned dirPerPart;
+        bool replicate;
+
+        // Standard file name with multiple parts
+        parseFileName("/var/lib/HPCCSystems/hpcc-data/test/myname._1_of_3", mname.clear(), num, max, stripeNum, dirPerPart, replicate);
+        CPPUNIT_ASSERT_EQUAL_STR("/var/lib/HPCCSystems/hpcc-data/test/myname._$P$_of_3",mname.str());
```

Comment:
trivial: (no need to change) - maybe slight overkill for new macro that converts each arg. to std::string's I think.
Could could be:
```
CPPUNIT_ASSERT(strsame("/var/lib/HPCCSystems/hpcc-data/test/myname._$P$_of_3", mname));
```